test/e2e/nodepool_test.go

```diff
@@ -44,9 +52,10 @@ func TestNodePool(t *testing.T) {
 	guestCluster := e2eutil.CreateCluster(t, ctx, mgmtClient, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
 	guestClient := e2eutil.WaitForGuestClient(t, ctx, mgmtClient, guestCluster)
 
-	// Set of tests
-	// Each test should have their own NodePool
-	t.Run("TestNodePoolAutoRepair", testNodePoolAutoRepair(ctx, mgmtClient, guestCluster, guestClient, clusterOpts))
+	// IMPORTANT: Ensure you updates the "testNodePoolCounter" with the right number of test to execute.
```

Comment:
Is this only needed because you are creating a common context for TestNodePool and closing it in that function?
Wouldn't the problem be solved if instead each `t.run(` instantiate its own ctx from the root one `testContext` i.e. `ctx, cancel := context.WithCancel(testContext)`?

Reply:
Nop, I've creating a context in each of the tests I have. I've tried that already :/

Reply:
I'd expect the outer caller i.e.  TestNodePool to just block until all subroutines return, so the context wouldn't be a problem https://pkg.go.dev/testing.
See examples there:

```
Subtests can also be used to control parallelism. A parent test will only complete once all of its subtests complete. In this example, all tests are run in parallel with each other, and only with each other, regardless of other top-level tests that may be defined:

func TestGroupedParallel(t *testing.T) {
    for _, tc := range tests {
        tc := tc // capture range variable
        t.Run(tc.Name, func(t *testing.T) {
            t.Parallel()
            ...
        })
    }
}
Run does not return until parallel subtests have completed, providing a way to clean up after a group of parallel tests:

func TestTeardownParallel(t *testing.T) {
    // This Run will not return until the parallel tests finish.
    t.Run("group", func(t *testing.T) {
        t.Run("Test1", parallelTest1)
        t.Run("Test2", parallelTest2)
        t.Run("Test3", parallelTest3)
    })
    // <tear-down code>
}
```

Reply:
It looks that it's the way to do it :). Thanks!